Common/include/toolboxes/C1DInterpolation.hpp

```diff
@@ -0,0 +1,167 @@
+/*!
+ * \file C1DInterpolation.hpp
+ * \brief Inlet_interpolation_functions
+ * \author Aman Baig
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation 
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <cmath>
+#include <vector>
+#include<fstream>
+#include "../datatype_structure.hpp"
+#include "../option_structure.hpp"
+
+using namespace std;
+
+class C1DInterpolation{
+protected:
```

Comment:
I suggest making the class members private where possible

Reply:
This class in inherited by CAkimaInterpolation and CLinearInterpolation and these variables are needed there.